knowledge-base/grid-handle-empty-popup-footer.md

```diff
@@ -0,0 +1,460 @@
+                            <FormItem Field="EmployeeId" Enabled="false"></FormItem>
+                            <FormItem Field="Name">
+                            </FormItem>
+                            <FormItem Field="HireDate" LabelText="Custom Hire Date Label"></FormItem>
+                            <FormItem>
+                                <Template>
+                                    <label for="position">Custom Position Label</label>
+                                    <TelerikDropDownList Data="@PositionsData"
+                                                         @bind-Value="@EditItem.Position"
+                                                         Id="position">
+                                    </TelerikDropDownList>
+                                </Template>
+                            </FormItem>
+                        </FormItems>
+                        <FormButtons>
+                            @*remove the deafult buttons from the Form*@
+                        </FormButtons>
+                    </TelerikForm>
+                }
+            </FormTemplate>
+            <ButtonsTemplate>
+                <TelerikButton Icon="@nameof(SvgIcon.Save)" OnClick="@OnSubmit">Save</TelerikButton>
+                <TelerikButton Icon="@nameof(SvgIcon.Cancel)" ButtonType="@ButtonType.Button" OnClick="@OnCancel">Cancel</TelerikButton>
+            </ButtonsTemplate>
+        </GridPopupEditFormSettings>
+    </GridSettings>
+    <GridColumns>
+        <GridColumn Field=@nameof(Person.EmployeeId) Editable="false" />
+        <GridColumn Field=@nameof(Person.Name) />
+        <GridColumn Field=@nameof(Person.HireDate) Title="Hire Date" />
+        <GridColumn Field=@nameof(Person.Position) Title="Position" />
+        <GridCommandColumn>
+            <GridCommandButton Command="Edit" Icon="@SvgIcon.Pencil">Edit</GridCommandButton>
+            <GridCommandButton Command="Delete" Icon="@SvgIcon.Trash">Delete</GridCommandButton>
+        </GridCommandColumn>
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private List<string> PositionsData { get; set; } = new List<string>()
+    {
+        "Manager", "Developer", "QA"
+    };
+
+    private TelerikGrid<Person> GridRef { get; set; }
+    private List<Person> GridData { get; set; }
+    private Person EditItem { get; set; }
+    private List<Person> _people;
+
+    public class Person
+    {
+        public int EmployeeId { get; set; }
+        public string Name { get; set; }
+        public DateTime HireDate { get; set; }
+        public string Position { get; set; }
+    }
+
+    public List<Person> People
+    {
+        get
+        {
+            if (_people == null)
+            {
+                _people = GeneratePeople(30);
+            }
+
+            return _people;
+        }
+    }
+
+    protected override void OnInitialized()
+    {
+        LoadData();
+    }
+
+    private void LoadData()
+    {
+        GridData = GetPeople();
+    }
```

Comment:
This looks over-engineered. Why not generate the data in OnInitialized and forget about it?